File: src/se/fredsberg/game/random/RandomScrambler.java
package se.fredsberg.game.random;

import java.util.Collections;
import java.util.List;

public class RandomScrambler implements Scrambler {

    @Override
    public int random(int upperExclusiveLimit) {
        return (int) (Math.random() * upperExclusiveLimit);
    }

    @Override
    public void scramble(List<?> values) {
        Collections.shuffle(values);
    }

}

File: jssrc/RandomScrambler.js
var BackBone = require('backbone');
var RandomScrambler = BackBone.Model.extend({
    random: function(upperExclusiveLimit) {
        return Math.floor(Math.random() * upperExclusiveLimit);
    },
    scramble: function(o) {
        for (var j, x, i = o.length; i; j = parseInt(Math.random() * i), x = o[--i], o[i] = o[j], o[j] = x);
    }
});
module.exports = RandomScrambler;